CMakeLists.txt

```diff
@@ -4,14 +4,36 @@
 
 message("CMakeLists.txt - NimbleSM")
 
-cmake_minimum_required(VERSION 3.12)
+# Needed to specify the version of NimbleSM
+CMAKE_POLICY(SET CMP0048 NEW)
+
 project(NimbleSM VERSION 0.1.0 LANGUAGES CXX)
 
+#--- Define options
+option(HAVE_UQ "Whether to use enamble UQ sampling" OFF)
+option(HAVE_BVH "Whether to use BVH for parallel asynchronous contact" OFF)
+option(HAVE_TRILINOS "Whether to use Trilinos" OFF)
+option(HAVE_KOKKOS "Whether to use Kokkos for parallel execution" OFF)
+option(USE_PURE_MPI "Whether to use Pure MPI (no Trilinos)" OFF)
+option(NIMBLE_ENABLE_UNIT_TESTS "Whether to build Nimble with GTest testing" OFF)
+option(TIME_CONTACT "Whether to time contact" OFF)
+option(HAVE_ARBORX "Whether to use ArborX" OFF)
+
+#--- Set CMake minimum version
+if (HAVE_KOKKOS OR HAVE_TRILINOS)
```

Comment:
Generally having inconsistent CMake versions is a bad idea because CMake will actually downgrade to the required version. This can cause some problems with any downstream projects. I seem to recall we agreed on updating the version to 3.17, so I don't think we need this unless there is a good reason for still allowing 3.12